go/store/nbs/store.go

```diff
@@ -1210,13 +1210,13 @@ func getTableFiles(css map[addr]chunkSource, contents manifestContents, numSpecs
 func newTableFile(cs chunkSource, info tableSpec) tableFile {
 	return tableFile{
 		info: info,
-		open: func(ctx context.Context) (io.ReadCloser, error) {
+		open: func(ctx context.Context) (io.ReadCloser, uint64, error) {
 			r, err := cs.reader(ctx)
 			if err != nil {
-				return nil, err
+				return nil, 0, err
 			}
 
-			return io.NopCloser(r), nil
+			return io.NopCloser(r), 0, nil
```

Comment:
Should this return the actual file size?

This is going to break reporting on `dolt clone file:///....`, but not sure that's a huge priority.

Reply:
I just added `size() (uint64, error)` to the `chunkSource` interface and then in the `tableReader` implementation, it returns the table file size as reported by the index.

I also updated the test cases to check that the reported size is correct.